server/server/Routes/signup.js

```diff
@@ -0,0 +1,22 @@
+const express = require("express");
+const router = express.Router();
+const { body } = require("express-validator");
+const userController = require("../controllers/userController");
+
+// Define the sign-up route
+router.post(
+  "/signup",
+
+  // Validation rules
+  body("username").isLength({ min: 5 }),
+  body("password").isLength({ min: 5 }),
+  body("email").isEmail(),
+  body("firstName").notEmpty(),
+  body("lastName").notEmpty(),
+  body("phoneNumber").isMobilePhone(),
```

Comment:
i think we should include a custom message with each validation rule to be sent in response, feel free to accept these changes:

```suggestion
  body("username", "Username should be at least 5 characters").isLength({ min: 5 }),
  body("password", "Password should be at least 5 characters").isLength({ min: 5 }),
  body("email", "Email should be in 'example@email.com' format").isEmail(),
  body("firstName", "First Name should not be empty").notEmpty(),
  body("lastName", "Last Name should not be empty").notEmpty(),
  body("phoneNumber", "Phone Number should be a Jordanian mobile number").isMobilePhone('ar-JO'),
```